apps/designer/app/dashboard/project-card.tsx

```
@@ -0,0 +1,295 @@
+        >
+          <MenuIcon width={15} height={15} />
+        </IconButton>
+      </DropdownMenuTrigger>
+      <DropdownMenuContent align="end">
+        <DropdownMenuItem onSelect={onDuplicate}>
+          <Text>Duplicate</Text>
+        </DropdownMenuItem>
+        <DropdownMenuItem onSelect={onRename}>
+          <Text>Rename</Text>
+        </DropdownMenuItem>
+        {isFeatureEnabled("share2") && (
+          <DropdownMenuItem>
+            <Text>Share</Text>
+          </DropdownMenuItem>
+        )}
+        <DropdownMenuItem onSelect={onDelete}>
+          <Text>Delete</Text>
+        </DropdownMenuItem>
+      </DropdownMenuContent>
+    </DropdownMenu>
+  );
+};
+
+const useProjectCard = () => {
+  const fetcher = useFetcher();
+  const designerLinkRef = useRef<HTMLAnchorElement>(null);
+
+  // @todo with dialog it can be displayed in the dialog
+  useEffect(() => {
+    if (fetcher.data?.errors) {
+      toast.error(fetcher.data.errors);
+    }
+  }, [fetcher.data]);
+
+  const handleKeyDown = (event: KeyboardEvent<HTMLElement>) => {
+    const elements: Array<HTMLElement> = Array.from(
+      event.currentTarget.querySelectorAll(`[tabIndex='-1']`)
+    );
+    const currentIndex = elements.indexOf(
+      document.activeElement as HTMLElement
+    );
+    switch (event.key) {
+      case "Enter": {
+        designerLinkRef.current?.click();
+        break;
+      }
+      case "ArrowUp":
+      case "ArrowRight": {
+        const nextElement = elements.at(currentIndex + 1) ?? elements[0];
+        nextElement?.focus();
+        break;
+      }
+      case "ArrowDown":
+      case "ArrowLeft": {
+        const nextElement = elements.at(currentIndex - 1) ?? elements[0];
+        nextElement?.focus();
+        break;
+      }
+    }
+  };
+  const { submit: deleteProject } = proxy.delete.useMutation();
+  const { submit: rename } = proxy.rename.useMutation();
+  const { submit: duplicate } = proxy.duplicate.useMutation();
```

Seems like we're ignoring errors from all of these